http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/LeoRoutesJsonCodec.scala

```diff
@@ -4,35 +4,25 @@ package api
 
 import cats.implicits._
 import io.circe.{Decoder, DecodingFailure}
+import org.broadinstitute.dsde.workbench.google2.MachineTypeName
 import org.broadinstitute.dsde.workbench.leonardo.JsonCodec._
-import org.broadinstitute.dsde.workbench.leonardo.http.service.{
-  CreateClusterAPIResponse,
-  GetClusterResponse,
-  ListClusterResponse,
-  UpdateClusterResponse
-}
-import org.broadinstitute.dsde.workbench.leonardo.model.LeonardoJsonSupport._
-import org.broadinstitute.dsde.workbench.leonardo.model.google.GoogleJsonSupport._
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, PropertyFilePrefix, RuntimeConfigRequest}
-import org.broadinstitute.dsde.workbench.model.WorkbenchIdentityJsonSupport._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.{CreateRuntimeRequest, RuntimeConfigRequest}
 import org.broadinstitute.dsde.workbench.model.google.GcsPath
-import org.broadinstitute.dsde.workbench.model.google.GoogleModelJsonSupport.{GcsPathFormat => _, _}
-import spray.json.{JsObject, _}
 
-object LeoRoutesJsonCodec extends DefaultJsonProtocol {
+object LeoRoutesJsonCodec {
   val invalidPropertiesError = DecodingFailure("invalid properties", List.empty)
 
   implicit val dataprocConfigDecoder: Decoder[RuntimeConfigRequest.DataprocConfig] = Decoder.instance { c =>
     for {
       numberOfWorkersInput <- c.downField("numberOfWorkers").as[Option[Int]]
-      masterMachineType <- c.downField("masterMachineType").as[Option[String]]
-      _ <- if (masterMachineType.nonEmpty && masterMachineType.exists(_.isEmpty)) Left(emptyMasterMachineType)
+      masterMachineType <- c.downField("masterMachineType").as[Option[MachineTypeName]]
+      _ <- if (masterMachineType.nonEmpty && masterMachineType.exists(_.value.isEmpty)) Left(emptyMasterMachineType)
```

Comment:
minor, this check probably can be moved to decoder for `MachineTypeName`

Reply:
True, will update